src/analysis/convolution.lean

```diff
@@ -1156,54 +1128,77 @@ begin
           by simp only [sub_add_cancel]⟩,
-        exact mem_closed_ball.2 hp.2.le, },
+        exact mem_closed_ball.2 hpx.2.le, },
       simp only [bound, indicator, this, if_true],
       refine mul_le_mul_of_nonneg_left _ (by positivity),
-      exact hε _ _ ⟨metric.mem_ball.2 hp.1.2, hp.1.1⟩ },
-    { have : g p (l p - a) = 0, from hgs _ _ hp.1.1 H,
+      exact hε _ _ ⟨metric.mem_ball.2 hpx.1.1, hpx.1.2⟩ },
+    { have : g p (x - a) = 0, from hgs _ _ hpx.1.2 H,
       simp only [this, bound, norm_zero, mul_zero],
       apply indicator_nonneg,
       assume a ha,
       positivity } },
   have I3 : integrable bound μ,
   { rw [integrable_indicator_iff hK'.measurable_set],
     exact ((hf hK').norm.const_mul _).mul_const _ },
-  have I4 : ∀ᵐ (a : G) ∂μ, continuous_within_at (λ q, L (f a) (g q (l q - a))) s q₀,
+  have I4 : ∀ᵐ (a : G) ∂μ, continuous_within_at (λ (q : P × G), L (f a) (g q.1 (q.2 - a)))
+    (s ×ˢ univ) q₀,
   { apply eventually_of_forall (λ a, _),
-    refine (L (f a)).continuous.continuous_at.comp_continuous_within_at _,
-    have : continuous_within_at (λ q, (q, l q - a)) s q₀,
-      from continuous_within_at_id.prod ((hl _ hq₀).sub continuous_within_at_const),
-    have h'q₀ : (q₀, l q₀ - a) ∈ s ×ˢ univ  := mk_mem_prod hq₀ (mem_univ _),
-    refine (show continuous_within_at (↿g) (s ×ˢ univ) ((λ q, (q, l q - a)) q₀),
-      from hg _ h'q₀).comp this (λ q hq, ⟨hq, mem_univ _⟩) },
+    suffices H : continuous_within_at (λ (q : P × G), (f a, g q.1 (q.2 - a))) (s ×ˢ univ) q₀,
+      from L.continuous₂.continuous_at.comp_continuous_within_at H,
+    apply continuous_within_at_const.prod,
+    change continuous_within_at (λ (q : P × G), ↿g (q.1, q.2 - a)) (s ×ˢ univ) q₀,
+    have : continuous_at (λ (q : P × G), (q.1, q.2 - a)) (q₀.1, q₀.2),
+      from (continuous_fst.prod_mk (continuous_snd.sub continuous_const)).continuous_at,
+    rw ← @prod.mk.eta _ _ q₀,
+    have h'q₀ : (q₀.1, q₀.2 - a) ∈ (s ×ˢ univ : set (P × G)) := ⟨hq₀, mem_univ _⟩,
+    refine continuous_within_at.comp (hg _ h'q₀) this.continuous_within_at _,
+    rintros ⟨q, x⟩ ⟨hq, hx⟩,
+    exact ⟨hq, mem_univ _⟩ },
   exact continuous_within_at_of_dominated I1 I2 I3 I4,
 end
 
+/-- The convolution `f * g` is continuous if `f` is locally integrable and `g` is continuous and
+compactly supported. Version where `g` depends on an additional parameter in an open subset `s` of
+a parameter space `P` (and the compact support `k` is independent of the parameter in `s`),
+given in terms of compositions with additional continuous maps. -/
+lemma continuous_on_convolution_right_with_param_comp {X : Type*} [topological_space X]
+  {u : X → P} {a : set X} (hu : continuous_on u a) {v : X → G} (hv : continuous_on v a)
+  {g : P → G → E'} {s : set P} (h'u : maps_to u a s) {k : set G}
+  (hk : is_compact k) (hgs : ∀ p, ∀ x, p ∈ s → x ∉ k → g p x = 0)
+  (hf : locally_integrable f μ) (hg : continuous_on (↿g) (s ×ˢ univ)) :
+  continuous_on (λ x, (f ⋆[L, μ] g (u x)) (v x)) a :=
```

Comment:
I think you can remove `u`, and just write `f ⋆[L, μ] g x` in the conclusion. 
If I then want to prove `continuous_on (λ x, (f ⋆[L, μ] g' (u x)) (v x)) a` in an application, I can just set `g x y := g' (u x) y`.
Removing `u` will also make it a lot easier for the elaborator to unify this lemma with an application, without giving `g` or `u` explicitly (or via other arguments).
EDIT: Same below. Also, without the `u` you don't have to cook up an arbitrary `u` in `has_compact_support.cont_diff_convolution_right`

Reply:
I am doubtful about that: if I have a lemma on continuity and I need to apply it to a composition, the library note you wrote suggests precisely to have a lemma that generalizes over all parameters because it is much more flexible in this way. It would feel awkward to generalize one variable but not the other one...

Reply:
For instance, in my application below, if I just use `g x`, then since the parameter space is just `unit` it means that I will have to take `x : unit` and I can't deduce anything out of it. On the other hand, with `g (u x)`, I can use as `u` any constant map into unit, and in particular I can take `x : G`, which is what I need.

Reply:
`g` is the parameter, and you generalize it by making it a binary function and replacing it with `g x`.
In your application below, `g_{this lemma} x y := g_{lemma below} y`, and there is no issue, and you don't have to cook up a random `u` yourself.

Reply:
To be more precise about parameters, here is the way I think about it. We are working with the function
`(g, v) \mapsto (f ⋆[L, μ] g) v` (we could also make the function depend on `f` or `L`, but we're not doing this in the current lemma). If we want to make the most general continuity/differentiability lemma, we are interested in making both `g` and `v` depend on a parameter `x`, so making `g` a binary function and `v` a unary function (it was a nullary function before) and then we are considering the map `x \mapsto (f ⋆[L, μ] g x) (v x)`.

Reply:
Thanks for the clarification. I'm convinced!